src/tests/system/tests/test_infopipe.py

```diff
@@ -410,3 +410,196 @@ def test_infopipe__lookup_user_attributes(client: Client, provider: LDAP):
     assert "mail" in user_attrs and user_attrs["mail"] == [
         "user1@example.com"
     ], "Expected mail to be ['user1@example.com']"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_infopipe__ping_interface(client: Client):
+    """
+    :title: Call the infopipe ping method
+    :setup:
+        1. Start SSSD
+    :steps:
+        1. Call ping method
+    :expectedresults:
+        1. Ping success
+    :customerscenario: False
+    """
+    client.sssd.start()
+
+    ifp = client.ifp.getObject("/org/freedesktop/sssd/infopipe")
+    result = ifp.Ping("ping")
+    assert result == "PONG", "Ping() did not return expected value"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_infopipe__list_components(client: Client):
+    """
+    :title: Call infopipe Listcomponents method
+    :setup:
+        1. Start SSSD
+    :steps:
+        1. Call ListComponents method
+        2. Verify returned components
+    :expectedresults:
+        1. Method returns list of components
+        2. At least monitor component is present
+    :customerscenario: False
+    """
+    client.sssd.start()
+
+    ifp = client.ifp.getObject("/org/freedesktop/sssd/infopipe")
+    components = ifp.ListComponents()
+    assert len(components) > 0, "No components returned"
+
+    component = client.ifp.getObject(components[0])
+    props = {}
+    properties_to_fetch = ["name", "type", "enabled", "debug_level"]
+    for prop in properties_to_fetch:
+        props[prop] = component.Get("org.freedesktop.sssd.infopipe.Components", prop)
+
+    assert "name" in props, "Component missing name property"
+    assert "type" in props, "Component missing type property"
+    assert "enabled" in props, "Component missing enabled property"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_infopipe__find_monitor(client: Client):
+    """
+    :title: Call infopipe FindMonitor method
+    :setup:
+        1. Start SSSD
+    :steps:
+        1. Call FindMonitor method
+        2. Get monitor properties
+    :expectedresults:
+        1. Returns monitor object path
+        2. Monitor properties contain expected values
+    :customerscenario: False
+    """
+    client.sssd.start()
+
+    ifp = client.ifp.getObject("/org/freedesktop/sssd/infopipe")
+    monitor_path = ifp.FindMonitor()
+    assert monitor_path.startswith("/org/freedesktop/sssd/infopipe/Components/"), "Invalid monitor path"
+
+    monitor = client.ifp.getObject(monitor_path)
+    props = {}
+    for prop in ["name", "type", "enabled"]:
+        props[prop] = monitor.Get("org.freedesktop.sssd.infopipe.Components", prop)
+
+    assert props["name"] == "monitor", f"Monitor name incorrect: got {props['name']}"
+    assert props["type"] == "monitor", f"Monitor type incorrect: got {props['type']}"
+    assert props["enabled"] is True, f"Monitor not enabled: got {props['enabled']}"
```

Comment:
Usually we also indicate the expected value in the assert message, this way we can check the differences

Reply:
Included the expected value in the assertion message